backend/tournesol/tests/test_text_search.py

```diff
@@ -0,0 +1,712 @@
+
+        Just check that the search still works.
+        """
+        word = "kook"
+        self._create_rated_entity(field1=word, language="et")
+
+        response = self.client.get(
+            self._make_url(word, "et"),
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_sorting_depends_on_total_score(self):
+        """
+        If the relevance is the same, the results with higher
+        total score (or "weighted score") should appear first.
+        """
+        scores = [.3, -4.7, 6.0, 0.0, 8.1, -10.0, -2.4, -2.3, 1.2, 1.1]
+        entities = []
+        for score in scores:
+            entities.append(self._create_rated_entity(field1=self.field, criteria_score=score))
+
+        entities = [self.setup_entity] + entities
+        scores = [self.setup_score] + scores
+        self.assertEqual(len(scores), len(entities))
+
+        response = self.client.get(self.url + "&unsafe=true", format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], len(scores))
+        self.assertEqual(len(response.data["results"]), response.data["count"])
+
+        sorted_scores = sorted(scores, reverse=True)
+        for index, score in enumerate(sorted_scores):
+            entity = entities[scores.index(score)]
+            self.assertEqual(response.data["results"][index]["uid"], entity.uid)
+
+    def test_public_recommendations_sorting_depends_on_total_score(self):
+        """
+        If the relevance is the same, the results with higher
+        total score (or "weighted score") should appear first.
+
+        Test it for public and private contributor recommendations
+        """
+        scores = [.3, -4.7, 6.0, 0.0, 8.1, -10.0, -2.4, -2.3, 1.2, 1.1]
+        entities = []
+        for score in scores:
+            entities.append(self._create_rated_entity(field1=self.field, criteria_score=score))
+
+        entities = [self.setup_entity] + entities
+        scores = [self.setup_score] + scores
+        self.assertEqual(len(scores), len(entities))
+
+        response = self.client.get(
+            f"/users/{self.user.username}/recommendations/{self.poll.name}"
+            f"?search={self.query}&metadata[language]={self.language}",
+            format="json"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], len(scores))
+        self.assertEqual(len(response.data["results"]), response.data["count"])
+
+        sorted_scores = sorted(scores, reverse=True)
+        for index, score in enumerate(sorted_scores):
+            entity = entities[scores.index(score)]
+            self.assertEqual(response.data["results"][index]["uid"], entity.uid)
+
+    def test_private_recommendations_sorting_depends_on_total_score(self):

```

Comment:
I'd add the term "personal" here too.

`def test_private_personal_reco_sorting_depends_on_total_score`